core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -136,8 +143,14 @@ public void onNext(T t) {
 
         @Override
         public void onError(Throwable t) {
-            beforeError(delegate, t);
-            delegate.onError(t);
+            final Throwable filteredCause = beforeError(delegate, t);
+            if (filteredCause != null) {
+                delegate.onError(filteredCause);
+            } else {
+                logger.warn("{}#beforeError() returned null. Using the original exception:",
+                            FilteredStreamMessage.this.getClass().getName(), t);
```

Comment:
I wonder if we should log the full trace here. We delegate the exception to `delegate` anyway, so it is likely that it will be handled there. It's probably OK to log `t.toString()` only?